coffee/src/CoffeeMachineLoop.java

```diff
@@ -0,0 +1,184 @@
+     *
+     * @param coffeeType Coffee type code (1 - espresso, 2 - latte, 3 - cappuccino).
+     *
+     * @return Has resources or not.
+     */
+    private static boolean hasResources(int coffeeType)
+    {
+        boolean has = true;
+
+        if(AMOUNT_OF_WATER < WATER_FOR_ONE_CUP[coffeeType]) {
+            System.out.println("Sorry, not enough water!");
+            has = false;
+        }
+
+        if(AMOUNT_OF_COFFEE < COFFEE_FOR_ONE_CUP[coffeeType]) {
+            System.out.println("Sorry, not enough coffee!");
+            has = false;
+        }
+
+        if(AMOUNT_OF_MILK < MILK_FOR_ONE_CUP[coffeeType]) {
+            System.out.println("Sorry, not enough milk!");
+            has = false;
+        }
+
+        if(AMOUNT_OF_CUPS == 0) {
+            System.out.println("Sorry, not enough cups!");
+            has = false;
+        }
+
+        return has;
+    }
+
+    private static void buy()
+    {
+        int coffeeType = readIntegerValue("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino:");
+        --coffeeType;
+
+        if (hasResources(coffeeType)) {
+            System.out.println("I have enough resources, making you a coffee!");
+
+            AMOUNT_OF_WATER -= WATER_FOR_ONE_CUP[coffeeType];
+            AMOUNT_OF_MILK -= MILK_FOR_ONE_CUP[coffeeType];
+            AMOUNT_OF_COFFEE -= COFFEE_FOR_ONE_CUP[coffeeType];
+            AMOUNT_OF_MONEY += MONEY_FOR_ONE_CUP[coffeeType];
+
+            AMOUNT_OF_CUPS--;
+        }
+    }
+
+    private static void fill()
+    {
+        AMOUNT_OF_WATER += readIntegerValue("Write how many ml of water do you want to add:");
+        AMOUNT_OF_MILK += readIntegerValue("Write how many ml of milk do you want to add:");
+        AMOUNT_OF_COFFEE += readIntegerValue("Write how many grams of coffee beans do you want to add:");
+        AMOUNT_OF_MONEY += readIntegerValue("Write how many disposable cups of coffee do you want to add:");
+    }
+
+    private static void take()
+    {
+        System.out.println("I gave you $" + AMOUNT_OF_MONEY);
+        AMOUNT_OF_MONEY = 0;
+    }
+
+    /**
+     * Reads an integer from standard input.
+     *
+     * @param msg Prompt message.
+     *
+     * @return Integer value.
+     */
+    private static int readIntegerValue(String msg)
+    {
+        System.out.println(msg);
+        System.out.print("> ");
+        return scan.nextInt();
+    }
+
+    /**
+     * Reads text from standard input.
+     *
+     * @param msg Prompt message.
+     *
+     * @return String value.
+     */
+    private static String readStringValue(String msg)
+    {
+        System.out.println(msg);
+        System.out.print("> ");
+        String val = scan.nextLine();
+
+        // ignores empty inputs
+        while(val.isEmpty()) {
+            val = scan.nextLine();
+        }
+        return val;
+    }
+
+    public static void main(String[] args)
+    {
+        setupMachine();
+
+        while (true) {
```

Comment:
@tatevikb while true is not good idea , please try to change it

Reply:
`while(true)` was replaced with ordinary while loop.